### ISSUE TITLE
Modify build processes to only run FVT daily

### ISSUE THREAD

**Issue by planetf1** [odpi/ci-management#78]
Currently our regular maven build takes around 10 minutes (on the build machine) to run some FVT tests. We expect to add more tests

For master on 'egeria' we have
 - A pre-commit verification build
 - A post-merge build
 - A scheduled daily build

We would like to skip the FVT tests on the pre-commit build to speed the time to commit permission,

The egeria code already offers the option to skip ie
mvn clean install-DskipOpenTypesTest

However if this approach looks feasible I would propose that we ie the egeria team add an option/rename to
-DskipFVT

as we would apply this to other tests as well.,

We would then need the pre-commit (in particular) build to pass this option ie
mvn clean install -DskipFVT

Is this reasonable?

An alternative is to invert the option ie to NOT run the tests by default, and explicitly run via
-DrunFVT
but if we do this, newbies (who don't know egeria) will skip tests when they experiment, and it's probably better the default is more rigourous..?

If the above sounds good, I'll raise an egeria issue to change the parameter used.

**Comment by schannamallu** [comment]
Hi @planetf1,
I would suggest -DskipFVT would be best option to skip the tests and we will add the this readme file as a instructions  for the newbies.

Please let me know after adding the -DskipFVT , we will create this by creating pre-commit verification build

**Comment by planetf1** [comment]
I've created a PR https://github.com/odpi/egeria/pull/1040

Adding '-DskipFVT' to the pre-commit verification build should be ok whether the PR is merged or not (any unused system parms will be ignored). This should improve the pre-commit build time noticeably.

**Comment by planetf1** [comment]
the above PR is now merged

**Comment by schannamallu** [comment]
Hi @planetf1 ,
Please elaborate more on this point `A pre-commit verification build`
I though odpi is using gerrit for build verification or developers follow the branching strategy ie *feature* branches and merge to master but NO strategy here. 

the precommit build should come in between  feature branch and master brach. Once build success Jenkins merge feature to master.
Please letme know your thoughts.

**Comment by planetf1** [comment]
This build -> https://jenkins.odpi.org/view/odpi-egeria/job/odpi-egeria-maven-verify-master-mvn35-openjdk8/

* We do not use gerrit
* developers work on their own forks & branches, and when ready create a pull request against egeria (mostly master)

**Comment by schannamallu** [comment]
@planetf1 , You want this change only in the 
- A pre-commit verification build

or is it ok to add to 
- A post-merge build
- A scheduled daily build

**Comment by planetf1** [comment]
Yes. I think we should leave in the others for now

**Comment by schannamallu** [comment]
@planetf1, Is it ok if we change for all the 3 builds.
We are using a common component (global-jjb) which will take the maven-goal and apply to all the 3 builds. So it's easy to implement the change for all 3 builds and with less build generation script changes.
But if we want to specify different maven-goals for each build, it will take some time and need some modifications in current build generation scripts and may result in some build related issues and fixing them.

CC: @tykeal , Can you share your thoughts.

**Comment by tykeal** [comment]
It's not too hard to modify what is currently happening. At present egeria is using the {project-name}-github-maven-jobs job group definition which defines all of the standard maven jobs. If you wish to override the verify job options you'll have to switch all of the projects that need this modification from the job group to listed all of the jobs in the group individually which will then allow you to specify a specific override for the build options on the verify, or any of the other jobs then.

To see the listing of all the jobs in the job group see our https://docs.releng.linuxfoundation.org/projects/global-jjb/en/latest/jjb/lf-maven-jobs.html#job-groups

**Comment by schannamallu** [comment]
@tykeal, Creating local job_group was the solution.
Added the local job_group and override the maven_goal for the verify job.

@planetf1, You can check the updated maven-goal for verify job once the PR is merged.

**Comment by planetf1** [comment]
Thanks. Looks good for the verify build which is now down to 13 minutes (the main saving we wanted).
I'll close - and if any other issues noted will open up new issue.
